File: web/client/src/app/model/account-transfer-export/internet-jpbank-exporter.ts
import { ExporterBase } from 'src/app/model/account-transfer-export/exporter-base'
import { AccountTransferDetail } from 'src/app/model/account-transfer-detail.model'

export class InternetJPBankExporter extends ExporterBase
{
    public GetHeaderRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionHeader),
            this.GetStrField(2,  this.PaymentAgency.contractCode),
            this.GetStrField(1,  this.Dummy()),
            this.GetNmbField(10, this.PaymentAgency.consigneeCode),
            this.GetStrField(40, this.Company.kana),
            this.GetStrField(4,  this.GetMMdd(this.DueAt)),
            this.GetNmbField(4,  this.PaymentAgency.bankCode),
            this.GetStrField(15, this.PaymentAgency.bankName),
            this.GetNmbField(3,  this.PaymentAgency.branchCode),
            this.GetStrField(16, this.Dummy()),
            this.GetNmbField(7,  this.PaymentAgency.accountNumber),
            this.GetStrField(4,  this.GetMMdd(this.DueAt2nd)),
            this.GetStrField(1,  this.Dummy()),
            this.GetStrField(12, this.Dummy())]);
    }

    public GetDataRecord(detail:AccountTransferDetail):string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionData),
            this.GetNmbField(4,  detail.transferBankCode),
            this.GetStrField(15, detail.transferBankName),
            this.GetNmbField(3,  detail.transferBranchCode),
            this.GetStrField(20, this.Dummy()),
            this.GetNmbField(7,  detail.transferAccountNumber),
            this.GetStrField(30, detail.transferAccountName),
            this.GetNmbField(10, parseInt(detail.billingAmount.toString())),
            this.GetStrField(1,  detail.transferNewCode),
            this.GetStrField(20, detail.transferCustomerCode),
            this.GetStrField(1,  0),
            this.GetStrField(4,  this.Dummy()),
            this.GetStrField(2,  this.Dummy()),
            this.GetStrField(2,  this.Dummy())]);
    }

    public GetTrailerRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1,  this.DivisionTrailer),
            this.GetNmbField(6,  this.TotalCount),
            this.GetNmbField(12, parseInt(this.TotalAmount.toString())),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(12, this.Dummy()),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(12, this.Dummy()),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(6,  this.Dummy()),
            this.GetNmbField(12, this.Dummy()),
            this.GetStrField(29, this.Dummy())]);
    }

    public GetEndRecord():string{
        return this.JoinFixedData([
            this.GetStrField(1, this.DivisionEnd),
            this.GetStrField(119, this.Dummy())]);
    }
}

File: src/Rac.VOne.AccountTransfer/Export/InternetJPBankExporter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Rac.VOne.Web.Models;


namespace Rac.VOne.AccountTransfer.Export
{
    public class InternetJPBankExporter : ExporterBase
    {
        public DateTime DueAt2nd { get; set; }

        public override string GetHeaderRecord() => Join(
             GetStrField( 1, DivisionHeader),
             GetStrField( 2, PaymentAgency.ContractCode),
             GetStrField( 1, Dummy()),
             GetNmbField(10, PaymentAgency.ConsigneeCode),
             GetStrField(40, Company.Kana),
             GetStrField( 4, DueAt.ToString("MMdd")),
             GetNmbField( 4, PaymentAgency.BankCode),
             GetStrField(15, PaymentAgency.BankName),
             GetNmbField( 3, PaymentAgency.BranchCode),
             GetStrField(16, Dummy()),
             GetNmbField( 7, PaymentAgency.AccountNumber),
             GetStrField( 4, DueAt2nd.ToString("MMdd")),
             GetStrField( 1, Dummy()),
             GetStrField(12, Dummy()));

        public override string GetDataRecord(AccountTransferDetail detail) => Join(
            GetStrField( 1, DivisionData),
            GetNmbField( 4, detail.TransferBankCode),
            GetStrField(15, detail.TransferBankName),
            GetNmbField( 3, detail.TransferBranchCode),
            GetStrField(20, Dummy()),
            GetNmbField( 7, detail.TransferAccountNumber),
            GetStrField(30, detail.TransferAccountName),
            GetNmbField(10, decimal.Truncate(detail.BillingAmount)),
            GetStrField( 1, detail.TransferNewCode),
            GetStrField(20, detail.TransferCustomerCode),
            GetStrField( 1, 0),
            GetStrField( 4, Dummy()),
            GetStrField( 2, Dummy()),
            GetStrField( 2, Dummy()));

        public override string GetTrailerRecord() => Join(
            GetStrField( 1, DivisionTrailer),
            GetNmbField( 6, TotalCount),
            GetNmbField(12, decimal.Truncate(TotalAmount)),
            GetNmbField( 6, Dummy()),
            GetNmbField(12, Dummy()),
            GetNmbField( 6, Dummy()),
            GetNmbField(12, Dummy()),
            GetNmbField( 6, Dummy()),
            GetNmbField( 6, Dummy()),
            GetNmbField( 6, Dummy()),
            GetNmbField( 6, Dummy()),
            GetNmbField(12, Dummy()),
            GetStrField(29, Dummy()));

        public override string GetEndRecord() => Join(
            GetStrField(  1, DivisionEnd),
            GetStrField(119, Dummy()));
    }
}
